rules/set-hostprocess-true/raw.rego

```diff
@@ -0,0 +1,41 @@
+package armo_builtins
+
+import future.keywords.if
+
+# Rule to check if container is set as a 'Host Process' container
+deny[msg] {
+    # Verify the object kind
+    obj := input[_]
+    allowed_kinds := {"Pod", "Deployment", "ReplicaSet", "DaemonSet", "StatefulSet", "Job", "CronJob"}
+    allowed_kinds[obj.kind]
+
+    # Check if HostProcess is set to true
+    not isHostProcessSet(obj)
+
+    path := sprintf("%v.spec.hostProcess", [obj.kind])
+    msg := {
+        "alertMessage": sprintf("%v: %v does not set 'spec.hostProcess' with allowed value", [obj.kind, obj.metadata.name]),
+        "packagename": "armo_builtins",
+        "alertScore": 7,
+        "failedPaths": [path],
+        "fixPaths": [],
+        "alertObject": {
+            "k8sApiObjects": [obj]
+        }
+    }
+}
+
+# Function to check if container is set as a 'Host Process' container
+isHostProcessSet(obj) := true if {
+    obj.spec.hostProcess == true
```

Comment:
You should check both for `PodSecurityContext` and `SecurityContext`. What I mean is that `windowsOptions` can be found both in these places, so you should check in both of them. You can find an example here: https://github.com/kubescape/regolibrary/blob/dev/rules/set-seLinuxOptions/raw.rego.
`seLinuxOptions` can be found in both the paths. So, the path variable that we are initializing, should have the right path according to which object has been checked.